src/containers/maps/MapExplorer.jsx

```diff
@@ -44,20 +50,37 @@ class MapExplorer extends React.Component {
         <MapVisuals
           mapConfig={this.props.mapConfig}
         />
-        <MapControls
-          mapConfig={this.props.mapConfig}
-        />
+        {(!this.props.activeCameraId)
+          ? <MapControls
+              mapConfig={this.props.mapConfig}
+            />
+          : <CameraViewer
+              mapConfig={this.props.mapConfig}
+              activeCameraData={this.props.activeCameraData}
+              activeCameraDataStatus={this.props.activeCameraDataStatus}
+              activeCameraMetadata={this.props.activeCameraMetadata}
+              activeCameraMetadataStatus={this.props.activeCameraMetadataStatus}
+            />
+        }
       </Box>
     );
   }
 }
 
 MapExplorer.propTypes = {
   mapConfig: PropTypes.object,
+  ...MAPEXPLORER_PROPTYPES,
 };
 MapExplorer.defaultProps = {
   mapConfig: null,
+  ...MAPEXPLORER_INITIAL_STATE,
 };
-const mapStateToProps = (state) => ({});
+const mapStateToProps = (state) => ({
+  activeCameraId: state.mapexplorer.activeCameraId,
+  activeCameraMetadata: state.mapexplorer.activeCameraMetadata,
```

Comment:
Since `CameraViewer` is a class, why not connect it to the redux store and directly set the props from the app state there? I think it's the only component using `activeCameraMetadata`, `activeCameraData` and the associated statuses, correct?

Reply:
Making CamerViewer connected to Redux makes sense, but my concept is that the main `MapExplorer` container should be responsible for the "overall/big picture" state of the Map feature.

That is to say, a new developer should be able to go into <MapExplorer /> and say, "Oh, right, that `if (activeCameraId)` clause means that `<CameraViewer />` is only available when an Active Camera is specified." Otherwise said dev would need to dig into CameraViewer to see that it renders null when activeCameraId is null.

Functionally, it's really the same, so if you feel strongly about making CameraViewer connected to Redux then I'll be happy to make the changes. My feelings for keeping it as is, is about 7/10.

Reply:
I think `activeCameraId` makes sense to remain in `MapExplorer` since it is handling the logic for the render. What I'm suggesting is the data that is specific to the `CameraViewer` directly be mapped fro the redux store, `activeCameraMetadata` and `activeCameraData`.

Just coming back from the React Rally conference and there's a lot we're not doing to improve performance in our apps at the moment (like correctly use `shouldComponentUpdate` for example). One suggestion was to make sure smart components only have access to the data they need.

 I don't feel too strongly about this at the moment, however, something to consider for future optimization.